addons/binding/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/service/MainTVServerService.java

```diff
@@ -176,7 +176,7 @@ public void onServiceSubscribed(String service, boolean succeeded) {
     public void onValueReceived(String variable, String value, String service) {
 
         String oldValue = stateMap.get(variable);
-        if (value.equals(oldValue)) {
+        if (oldValue.equals(value)) {
```

Comment:
Yes, agreed. But exactly because of that wasn't the previous solution the right one? With "Since value is always non-null" and "oldValue can potentially be null" you have introduced `null.equals(String)` which *will* raise the NPE...!? Pardon me if I'm missing something obvious